guides/micronaut-data-jdbc-repository/kotlin/src/main/kotlin/example/micronaut/GenreRepository.kt

```diff
@@ -0,0 +1,24 @@
+package example.micronaut
+
+import example.micronaut.domain.Genre
+import io.micronaut.data.annotation.Id
+import io.micronaut.data.exceptions.DataAccessException
+import io.micronaut.data.jdbc.annotation.JdbcRepository
+import io.micronaut.data.model.query.builder.sql.Dialect
+import io.micronaut.data.repository.PageableRepository
+import javax.transaction.Transactional
+import javax.validation.constraints.NotBlank
+
+@JdbcRepository(dialect = Dialect.MYSQL) //<1>
+open abstract class GenreRepository : PageableRepository<Genre, Long> { //<2>
```

Comment:
can use use an interface where `saveWithException` is a Kotlin method with default implementation annotated with `@JvmDefault`?

Reply:
`@JvmDefault` is deprecated and is only allowed with -Xjvm-default option. Therefore don't think it is a valid option.